application.py

```diff
@@ -440,8 +450,9 @@ def contest_problem(contest_id, problem_id):
     # Reached via POST
 
     # Ensure contest hasn't ended
-    end = datetime.strptime(check1[0]["end"], "%Y-%m-%d %H:%M:%S")
-    if datetime.now() > end:
+    end = db.execute("SELECT end FROM contests WHERE id=:id", id=contest_id)
+    end = datetime.strptime(end[0]["end"], "%Y-%m-%d %H:%M:%S")
+    if datetime.utcnow() > end:
```

Comment:
Initial review: see below
From the Python docs:
Warning Because naive datetime objects are treated by many datetime methods as local times, it is preferred to use aware datetimes to represent times in UTC. As such, the recommended way to create an object representing the current time in UTC is by calling datetime.now(timezone.utc).